vita3k/modules/include/modules/tracy.h

```diff
@@ -28,6 +30,36 @@ std::string to_debug_str(const MemState &mem, T data) {
     return datass.str();
 }
 
+// Override pointers, we want to print the address in hex
+template <typename U>
+std::string to_debug_str(const MemState &mem, U *data) {
+    std::stringstream datass;
+    datass << log_hex_fast((uint64_t)data); // Cast the pointer address to a 64 bit uint
+    return datass.str();
+}
+
+// Override for guest pointers, we want to print the host address instead
+template <typename U>
+std::string to_debug_str(const MemState &mem, Ptr<U> data) {
+    std::stringstream datass;
+    datass << log_hex_fast((uint64_t)data.get(mem));
+    return datass.str();
+}
+
+// Override for char pointers as the contents are readable
+template <typename U>
+std::string to_debug_str(const MemState &mem, char *data) {
+    // Format for correct char* should be "(address in hex (0x12345)) (string)", this is just in the
+    // extreme case that the string is actually "0x0 NULLPTR" and be confusing
+    return std::string(data ? log_hex_fast(data) + " " + data : "0x0 NULLPTR");
+}
+
+// Override for char pointers as the contents are readable
+template <typename U>
+std::string to_debug_str(const MemState &mem, const char *data) {
+    return std::string(data ? log_hex_fast(data) + " " + data : "0x0 NULLPTR");
+}
+
```

Comment:
I don't know what is better: unused <typename U> or inline

Reply:
inline, turns out the other way doesn't work and  the template gets ignored, now it should work as expected
![image](https://user-images.githubusercontent.com/30161277/194712825-228b4cd2-d764-4673-94b3-c89f9466d404.png)
![image](https://user-images.githubusercontent.com/30161277/194712829-974d5330-3046-4dff-a4eb-c5e57efe2fb8.png)
![image](https://user-images.githubusercontent.com/30161277/194712835-3101a6f5-007e-40f9-ba6a-b253b5b58866.png)